source/MaterialXShaderGen/ShaderGenerators/Glsl/GeomAttrValueGlsl.cpp

```diff
@@ -0,0 +1,60 @@
+#include <MaterialXShaderGen/ShaderGenerators/Glsl/GeomAttrValueGlsl.h>
+
+namespace MaterialX
+{
+
+SgImplementationPtr GeomAttrValueGlsl::creator()
+{
+    return std::make_shared<GeomAttrValueGlsl>();
+}
+
+void GeomAttrValueGlsl::createVariables(const SgNode& node, ShaderGenerator& shadergen, Shader& shader_)
+{
+    HwShader& shader = static_cast<HwShader&>(shader_);
+
+    const SgInput* attrNameInput = node.getInput(ATTRNAME);
+    if (!attrNameInput)
+    {
+        throw ExceptionShaderGenError("No 'attrname' parameter found on geomattrvalue node '" + node.getName() + "', don't know what attribute to bind");
+    }
+    const string name = attrNameInput->value->getValueString();
+    const string type = shadergen.getSyntax()->getTypeName(node.getOutput()->type);
+    const string variable = "geomattr_" + name;
```

Comment:
I guess "geomattr_" is added to avoid name clashes? This will require a bit of parsing to strip out the i_geomattr_ to get the actual input name.

Reply:
Yes, it's to avoid collisions. I'm not sure there's a way around that though because the "i_" needs to be stripped anyway. But when binding a variable you need to match against all known variables anyway, and if no match is found then you can look for "i_geomattr_" (or "u_geomattr_" since I will change this to a uniform now)